rules/aip0203/field_behavior_required.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package aip0203
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var requiredOneOfFieldBehavior = stringset.New(
+	"OPTIONAL", "REQUIRED", "OUTPUT_ONLY",
+)
+
+var fieldBehaviorRequired = &lint.FieldRule{
+	Name: lint.NewRuleName(203, "field-behavior-required"),
```

Comment:
I kind of liked the guidance in AIP-203 for OUTPUT_ONLY:
> Notably, fields in response messages (a message which only ever acts as a return value to an RPC, usually 
> ending in Response) **should not** be described as output only because this is already implied.

Do you think that is worth incorporating in this rule? e.g. only lint fields that are used in the context of a request. I realize that will complicate the `OnlyIf`, but I want your take.

Reply:
ac k and agreed, I'll filter out responses.

Reply:
FWIW the scope of a "response only" message is larger than just "messages that end with Response". Not sure how deep/broad you want to go here, but any nested message in a response (or lro metadata) message would be in scope of the exception.

Reply:
yeah I just realized that. 

The only way I can think of is another service message that looks at RPCs ->
requests, then iterate through messages that way. thoughts?

This rule is going to be pretty loud (not noisy, it's very clear feedback) so I agree with reducing
the cases where it runs here.